source/common/http/conn_manager_impl.cc

```diff
@@ -263,13 +263,23 @@ void ConnectionManagerImpl::doDeferredStreamDestroy(ActiveStream& stream) {
   if (connection_idle_timer_ && streams_.empty()) {
     connection_idle_timer_->enableTimer(config_.idleTimeout().value());
   }
+  if (requests_overflow_ && streams_.empty()) {
+    doConnectionClose(Network::ConnectionCloseType::FlushWrite, absl::nullopt, "");
+  }
 }
 
 RequestDecoder& ConnectionManagerImpl::newStream(ResponseEncoder& response_encoder,
                                                  bool is_internally_created) {
   if (connection_idle_timer_) {
     connection_idle_timer_->disableTimer();
   }
+  accumulated_requests_++;
+  if (config_.maxRequestsPerConnection() > 0 &&
+      accumulated_requests_ >= config_.maxRequestsPerConnection()) {
+    read_callbacks_->connection().readDisable(true);
```

Comment:
This will result in a non-graceful termination of the connection, and probably won't allow the outstanding requests to complete.

When the threshold is reached, I think you need to call `startDrainSequence()`. 

/wait

Reply:
Done.